src/Infrastructure/Repositories/LocalFilesRepository.php

```diff
@@ -42,84 +42,107 @@ public function getDefaultDirectory(): string
         return (string) getcwd();
     }
 
+    /**
+     * @return array<string, \Symfony\Component\Finder\SplFileInfo>
+     */
     public function getFiles(): array
     {
-        if ($this->files === null) {
-            $this->files = $this->getFilesList();
+        if (count($this->files) !== 0) {
+            return $this->files;
         }
 
-        return $this->files;
+        $withPathInfo = static function (string $path): array {
+            return [
+                'dirname' => $dirname = pathinfo($path, PATHINFO_DIRNAME),
+                'basename' => $basename = pathinfo($path, PATHINFO_BASENAME),
+                'full_path' => $dirname . DIRECTORY_SEPARATOR . $basename,
+                'is_file' => ! is_dir($path) && is_file($path),
+            ];
+        };
+
+        $paths = array_map($withPathInfo, $this->paths);
```

Comment:
Previously, if we called the `getFiles()` function without calling the `within` function, it's returned all the files in current working directory, right ? 
And now, if we call it, as `$this->paths` is initialized with an empty array, it should return nothing.

:sweat_smile: However, we actually don't call getFiles() without calling `within()` before, but maybe should we throw an exception if it's the case ? 

WDYT @olivernybroe ?

Reply:
This is tricky because from a pure code quality perspective that is a bad pattern ("temporal coupling" is the term I've heard most for it): the use of a public interface should not require knowledge of the internals, it should not be necessary to call method `a` before method `b`.

The fix here should probably be a better default case: the old behaviour you're talking about -- current working directory used if no directory has been provided -- could be added as an explicit default when `within()` hasn't been called. I'll try and restore the old behaviour with a better default case -- hopefully I can find a clean way to do it! I'll let you know once I've pushed that change.

Reply:
Yeah, it was just a reflection. I am not sure what is the best. 

Asking getFiles() without specify a directory could return all files as well as return no file. I really don't know what should be the correct behaviour...